iceberg-rest-server/src/test/java/com/datastrato/gravitino/iceberg/integration/test/util/IcebergRESTServerManager.java

```diff
@@ -0,0 +1,167 @@
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public abstract class IcebergRESTServerManager {
+
+  protected static final Logger LOG = LoggerFactory.getLogger(IcebergRESTServerManager.class);
+
+  protected Map<String, String> customConfigs = new HashMap<>();
+  protected Config serverConfig;
+  protected String checkUri;
+
+  public abstract Path getConfigDir();
+
+  public abstract Optional<Future> doStartIcebergRESTServer() throws Exception;
+
+  public abstract void doStopIcebergRESTServer();
+
+  public static IcebergRESTServerManager create() {
+    String testMode =
+        System.getProperty(com.datastrato.gravitino.integration.test.util.ITUtils.TEST_MODE);
+    if (com.datastrato.gravitino.integration.test.util.ITUtils.EMBEDDED_TEST_MODE.equals(
+        testMode)) {
+      return new IcebergRESTServerManagerForEmbedded();
+    } else {
+      return new IcebergRESTServerManagerForDeploy();
+    }
+  }
+
+  public void registerCustomConfigs(Map<String, String> configs) {
+    customConfigs.putAll(configs);
+  }
+
+  public Config getServerConfig() {
+    return serverConfig;
+  }
+
+  public void startIcebergRESTServer() throws Exception {
+    initServerConfig();
+    Optional<Future> future = doStartIcebergRESTServer();
+
+    long beginTime = System.currentTimeMillis();
+    boolean started = false;
+
+    while (System.currentTimeMillis() - beginTime < 1000 * 60) {
+      started = isHttpServerUp(checkUri);
+      if (started || (future.isPresent() && future.get().isDone())) {
+        break;
+      }
+      sleepUninterruptibly(500, TimeUnit.MILLISECONDS);
+    }
+    if (!started) {
+      try {
+        if (future.isPresent()) {
+          future.get().get(1, TimeUnit.SECONDS);
+        }
+      } catch (Exception e) {
+        throw new RuntimeException("IcebergRESTServer start failed", e);
+      }
+      throw new RuntimeException("Can not start IcebergRESTServer");
+    }
+  }
+
+  public void stopIcebergRESTServer() {
+    doStopIcebergRESTServer();
+    sleepUninterruptibly(500, TimeUnit.MILLISECONDS);
+
+    long beginTime = System.currentTimeMillis();
+    boolean started = true;
+    while (System.currentTimeMillis() - beginTime < 3 * 1000 * 60) {
+      sleepUninterruptibly(500, TimeUnit.MILLISECONDS);
+      started = isHttpServerUp(checkUri);
+      if (!started) {
+        break;
+      }
+    }
+    if (started) {
+      throw new RuntimeException("Can not stop IcebergRESTServer");
+    }
+  }
+
+  private static boolean isHttpServerUp(String testUrl) {
+    try (CloseableHttpClient httpClient = HttpClients.createDefault()) {
+      HttpGet request = new HttpGet(testUrl);
+      ClassicHttpResponse response = httpClient.execute(request, a -> a);
+      return response.getCode() == 200;
+    } catch (Exception e) {
+      LOG.warn("Check IcebergRESTServer failed, url:{}, error:{} ", testUrl, e.getMessage());
```

Comment:
Since the method name is `isHttpServerUp`, the error messages here should be not `IcebergRESTServer`.

Reply:
done